src/NUnitFramework/framework/Internal/PropertyNames.cs

```diff
@@ -137,6 +137,11 @@ public class PropertyNames
         /// The optional Order the test will run in
         /// </summary>
         public const string Order = "Order";
+
+        /// <summary>
+        /// The attempt currently being executed in a test using the <see cref="RetryAttribute"/>
+        /// </summary>
+        public const string CurrentAttempt = "CurrentAttempt";
```

Comment:
When used in the code of `RetryAttribute` the name `PropertyNames.CurrentAttempt` makes sense. When accessed outside of the attribute, however, it's a bit ambiguous. Current attempt to do what? I suggest "CurrentRetryCount".

Reply:
_CurrentRetryCount_ would be 0 on the first execution of the test as I would understand it though right (since it hasn't yet been retried)? Would _CurrentTestExecutionIteration_ work for you or do you want to keep the association of _RetryAttribute_ with _CurrentRetryCount_?

Reply:
Good point. How does `CurrentRepeatCount` seem to you? It could be set by both the Retry and the Repeat attributes. A value of zero would indicate that no repetition was in progress - i.e. neither attribute has been used. I can imagine `Repeat` and `Retry` being merged at some time in the future since `Retry` is just `Repeat` with a particular rule for when to stop.

Reply:
While `CurrentRepeatCount` as a Property is not my favourite because, to me it doesn't seem as clear as `CurrentTestExecutionIteration` (also not great as it is quite a mouthful), I've updated the code to use `CurrentRepeatCount` and to expose the value in the _TestContext_ by way of the _TestExecutionContext_. Hopefully this change is as you wanted. 😄

Reply:
Well, I only asked if you liked it! 😄 

`CurrentTestExectionIteration` sounds (to me) like we are re-running __all__ the tests, not just one, although I can see how it could mean that too. `CurrentRepeatCount` could be similarly ambiguous, I admit, but at least it matches the existing `Repeat` saved in the test by the `RepeatAttribute`. `CurrentRepetition` might work as well if you want something shorter.

As you have it now is basically as I imagined it. @jnm2 since you have been looking at this would you care to review too?